.github/workflows/build_pyvelox.yml

```diff
@@ -0,0 +1,177 @@
+      ref:
+        description: 'git ref to build'
+        required: false
+      publish:
+        description: 'publish to PyPI'
+        required: false
+        type: boolean
+        default: false
+  schedule:
+    - cron: '15 0 * * *'
+  pull_request:
+    paths:
+      - 'velox/**'
+      - '!velox/docs/**'
+      - 'third_party/**'
+      - 'pyvelox/**'
+      - '.github/workflows/build_pyvelox.yml'
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.repository }}-${{ github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, macos-11]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref || github.ref }}
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+    
+      - name: "Determine Version"
+        if: ${{ !inputs.version && github.event_name != 'pull_request' }}
+        id: version
+        run: |
+          # count number of commits since last tag matching a regex
+          # and use that to determine the version number
+          # e.g. if the last tag is 0.0.1, and there have been 5 commits since then
+          # the version will be 0.0.2.dev5
+          git fetch --tags
+          INITIAL_COMMIT=5d4db2569b7c249644bf36a543ba1bd8f12bf77c
+          BASE_VERSION=0.0.0
+
+          LAST_TAG=$(git describe --tags --match "pyvelox-v[0-9]*" --abbrev=0 || echo $INITIAL_COMMIT)
+          COMMITS_SINCE_TAG=$(git rev-list --count ${LAST_TAG}..HEAD)
+
+          if [ "$LAST_TAG" = "$INITIAL_COMMIT" ]; then
+            VERSION=$BASE_VERSION
+          else
+            VERSION=$(echo $LAST_TAG | sed '/pyvelox-v//')
+          fi
+          NEXT_VERSION=$(echo $VERSION | awk -F. -v OFS=. '{$NF++ ; print}')
+          echo "build_version=${NEXT_VERSION}.dev${COMMITS_SINCE_TAG}" >> $GITHUB_OUTPUT
+          
+      - run: mkdir -p .ccache
+      - name: "Restore ccache"
+        uses: actions/cache/restore@v3
+        id: restore-cache
+        with:
+          path: ".ccache"
+          key: ccache-wheels-${{ matrix.os }}-${{ github.sha }}
+          restore-keys: |
+            ccache-wheels-${{ matrix.os }}-
+
+      - if: matrix.os == 'macos-11'
+        run: |
+          echo "OPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1/" >> $GITHUB_ENV
+
+      - name: "Create sdist"
+        if: matrix.os == 'ubuntu-22.04'
+        env:
+          BUILD_VERSION: "${{ inputs.version || steps.version.outputs.build_version }}"
+        run: |
+          python setup.py sdist --dist-dir wheelhouse
+                  
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.12.0
+        env:
+          # required for preadv/pwritev
+          MACOSX_DEPLOYMENT_TARGET: "11.0"
```

Comment:
nit: Is this why wheels would not install ? Can we set it to 10.9 to be more backwards compatible (?)

Reply:
No due to the use of `preadv` and `pwritev`  11.0 is the lowest possible deployment target.